torch_frame/data/dataset.py

```diff
@@ -87,12 +90,13 @@ def canonicalize_col_to_pattern(col_to_pattern: Union[Optional[Any],
         for col in columns:
             col_to_pattern[col] = pattern
     else:
-        missing_cols = set(columns) - set(col_to_pattern.keys())
-        if len(missing_cols) > 0:
+        if all_inclusive and len(set(columns) -
```

Comment:
It might worth adding a quick test case :)

Reply:
Added.